bubblesort.rb

```diff
@@ -0,0 +1,45 @@
+def bubble_sort(array)
+  n = array.length - 1
+  loop do
+    swapped = false
+    for i in 0..n - 1
+      if array[i] > array[i + 1]
+        array[i], array[i + 1] = array[i + 1], array[i]
+        swapped = true
+      end  
+    end
+    n -= 1
+    break unless swapped
+  end
+  array
+end
+
+def bubble_sort_by(array)
+  n = array.length - 1
+  loop do
+    swapped = false
+    for i in 0..n - 1
+      if block_given?
+        if yield(array[i],array[i + 1]) > 0
+          array[i], array[i + 1] = array[i + 1], array[i]
+          swapped = true
+        end
+      elsif array[i].length > array[i + 1].length
```

Comment:
this is good for covering the strings, but what if it's the number then, this wouldn't work. Since we are talking about the enumerable class of ruby , focus is on sorting the numbers, even when you are sorting the strings you are sorting them by numbers (their respective length) `left.length - rigth.length` will actually yield a number.